### DISCUSSION TITLE
Need for RelayBandwidthRate and RelayBandwidthBurst?

### DISCUSSION THREAD

**Discussion by sreknob** [Ilshidur/tor-relay-docker#12] [General]
Hi - thanks so much for the update to the latest tor version! It's working nicely.

I was wondering, is there a reason that the container ENV defaults to 
```
ENV TOR_RelayBandwidthRate "100 KBytes"
ENV TOR_RelayBandwidthBurst "200 KBytes"
```
rather than only specifying them in the run command?
Tor's defaults, AFAIK is 0 for both of these.
I'm aware that I can set them to 0 as a variable, just curious :-)

**Comment by Ilshidur** [comment]
Hello sreknob.

I guess the reason why I did this back then was to make this image easily runnable for test purposes, so it would not need all those environment variables to run in an effective and non-invasive way. This way, the user running a container from this image just of the sake of trying it out won't risk having all their bandwidth being eaten up in case they forget it. ¯\_(ツ)_/¯
Now that I'm thinking about it, I realize this is a quite low limitation, even for testing.

**Comment by Ilshidur** [comment]
Moved this issue to a discussion.

**Comment by sreknob** [comment]
Thanks, discussion is better!
I agree that it's great to have a low threshold for testing, and that it doesn't necessarily need to change.
Perhaps just calling it out in the ReadMe that this value is automatically set and that you need to specify these variables if you want it to be anything different. I know that you have them listed in the docker run, but I would assume that if you left them out that it would be at the defaults for Tor, rather than the specified value.
Also, in the interest of minimizing variables to pass, I tried to pass my own custom torrc file but had issues with it overwriting still with the defaults from the ENV set in the docker build. Looking at your entrypoint file, I set my RELAY_TYPE to "custom" and mapped torrc file to "torr.custom.default" but it seemed to me that the default ENV variables still overwrote this. I guess it's because the torrc file that it merges with the defaults which build out first and overrides my torr.custom.default file.
Just some food for thought and again, thanks for this, I appreciate your work - I have it up and running nicely at 0.4.7.7.

**Comment by Ilshidur** [comment]
Closing thanks to @Likogann's work on https://github.com/Ilshidur/tor-relay-docker/pull/18.
